dapp/src/components/page/proposal/ProposalPage.tsx

```diff
@@ -0,0 +1,130 @@
+import React from "react";
+import { useState, useEffect } from "react";
+import { useStore } from "@nanostores/react";
+import ProposalPageTitle from "./ProposalPageTitle";
+import ProposalStatusSection from "./ProposalStatusSection";
+import VoteStatusBar from "./VoteStatusBar";
+import VotersModal from "./VotersModal";
+import VotingModal from "./VotingModal";
+import { connectedPublicKey, proposalId } from "utils/store";
+import type { VoteType, VoteStatus, ProposalStatus } from "types/proposal";
+import { demoProposalData } from "constants/demoProposalData";
+import ProposalDetail from "./ProposalDetail";
+import {
+  fetchOutcomeDataFromIPFS,
+  fetchProposalFromIPFS,
+} from "@service/ProposalService";
+import type { ProposalOutcome } from "types/proposal";
+import ExecuteProposalModal from "./ExecuteProposalModal";
+
+const ProposalPage: React.FC = () => {
+  const id = useStore(proposalId);
+  const connectedAddress = useStore(connectedPublicKey);
+  const [isVotersModalOpen, setIsVotersModalOpen] = useState(false);
+  const [isVotingModalOpen, setIsVotingModalOpen] = useState(false);
+  const [isExecuteProposalModalOpen, setIsExecuteProposalModalOpen] =
+    useState(false);
+  const [maintainers, setMaintainers] = useState<string[]>([]);
+  const [description, setDescription] = useState("");
+  const [outcome, setOutcome] = useState<ProposalOutcome | null>(null);
+  const [voteType, setVoteType] = useState<VoteType>();
+  const [voteStatus, setVoteStatus] = useState<VoteStatus>();
+  const [proposalStatus, setProposalStatus] = useState<ProposalStatus | null>(
+    null,
+  );
+  const [endDate, setEndDate] = useState<string | null>(null);
+
+  const openVotersModal = (voteType: VoteType) => {
+    if (proposalStatus !== "active") {
+      setVoteType(voteType);
+      setIsVotersModalOpen(true);
+    } else {
+      alert("Cannot show voters while voting is in progress");
+    }
+  };
+
+  const openVotingModal = () => {
+    if (proposalStatus === "active") {
+      if (connectedAddress) {
+        setIsVotingModalOpen(true);
+      } else {
+        alert("Please connect your wallet first");
+      }
+    } else {
+      alert("Cannot vote anymore, voting is ended.");
```

Comment:
We should not be able to see the button nor click on it, so no need to alert no?